webapp/graphite/storage.py

```diff
@@ -331,6 +371,127 @@ def distance_to_requested_interval(node):
             reader = MultiReader(minimal_node_set)
             return LeafNode(path, reader)
 
+    def tagdb_auto_complete_tags(self, exprs, tagPrefix=None, limit=None, requestContext=None):
+        log.debug(
+            'graphite.storage.Store.auto_complete_tags :: Starting lookup on all backends')
+
+        if requestContext is None:
+          requestContext = {}
+
+        jobs = []
+        use_tagdb = False
```

Comment:
part of this code seems duplicated between tags and value

Reply:
They're broadly similar, but there are a lot of small differences (log messages, job parameters).  I could try to abstract it away into a helper but I'm not sure how complex it would get trying to handle the differences.